reefperf/node_connection_provider.py

```diff
@@ -0,0 +1,54 @@
+from abc import abstractmethod, ABCMeta
+from paramiko import SSHClient, AutoAddPolicy
+from paramiko.rsakey import RSAKey
+from paramiko.client import MissingHostKeyPolicy
+from io import StringIO
+
+
+class NodeConnection(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def exec_command(self, command_str):
+        pass
+
+    @abstractmethod
+    def scp_file(self, path_to_file):
+        pass
+
+
+class ParamikoConnection(NodeConnection):
+    KEY_SIZE = 4096
+    KEY_CLASSES = {
+        'rsa': RSAKey
+    }
+
+    def __init__(self, conn_cfg, ipv4, username):
+        priv_key_type = conn_cfg['keys']['keys_type']
+        priv_key_str = conn_cfg['keys']['priv_key_str']
+        priv_key_stream = StringIO(priv_key_str)
+        priv_key = self.KEY_CLASSES[priv_key_type].from_private_key(
+            priv_key_stream
+        )
+        self._client = SSHClient()
+        self._client.set_missing_host_key_policy(AutoAddPolicy())
+        self._client.connect(ipv4, username=username, pkey=priv_key)
```

Comment:
that should not be called `ipv4`. it should work fine with `ipv6` and even domain names. `hostname`.